src/dolphin/checkVersion.ts

```diff
@@ -0,0 +1,63 @@
+import { ApolloClient, gql, HttpLink, InMemoryCache } from "@apollo/client";
+import { isDevelopment } from "common/constants";
+import { fetch } from "cross-fetch";
+import { GraphQLError } from "graphql";
+
+import { DolphinLaunchType, DolphinVersionResponse } from "./types";
+
+const httpLink = new HttpLink({ uri: process.env.SLIPPI_GRAPHQL_ENDPOINT, fetch });
+
+const appVersion = __VERSION__;
+
+const client = new ApolloClient({
+  link: httpLink,
```

Comment:
Why do we need to provide a link here instead of just providing the `uri` directly? Does it not work if we don't provide it an instance to `cross-fetch`?

Reply:
Yep, we get this explicit error from ApolloClient

```
Invariant Violation:
  "fetch" has not been found globally and no fetcher has been configured. To fix this, install a fetch package (like https://www.npmjs.com/package/cross-fetch), instantiate the fetcher, and pass it into your HttpLink constructor. For example:

  import fetch from 'cross-fetch';
  import { ApolloClient, HttpLink } from '@apollo/client';
  const client = new ApolloClient({
    link: new HttpLink({ uri: '/graphql', fetch })
  });
  ```